server/src/main/java/org/elasticsearch/transport/TransportHandshaker.java

```diff
@@ -257,6 +262,19 @@ static void ensureCompatibleVersion(
         Object channel
     ) {
         if (TransportVersion.isCompatible(remoteTransportVersion)) {
+            if (remoteTransportVersion.before(V8_18_HANDSHAKE_VERSION)) {
+                deprecationLogger.warn(
+                    DeprecationCategory.OTHER,
+                    "handshake_version",
+                    "Performed a handshake with a remote node whose version will be incompatible after this node "
+                        + "is upgraded to 9.x. This is likely a result of a CCR/CCS request. "
+                        + "Remote version: [{}]. "
+                        + "This version: [{}]. "
```

Comment:
Reporting the versions probably isn't all that helpful, we should report `channel` too so that users have more chance of identifying the node in question.